tests/unit/schema/test_schema_validation.py

```diff
@@ -0,0 +1,60 @@
+from pathlib import Path
+import jsonschema
+from parameterized import parameterized
+from unittest import TestCase
+
+from samcli.lib.config.file_manager import FILE_MANAGER_MAPPER
+from schema.make_schema import generate_schema
+
+
+class TestSchemaValidation(TestCase):
+    schema = None
+    testdata_dir = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.schema = generate_schema()
+        testing_dir = Path(__name__).resolve().parents[0]
+        cls.testdata_dir = str(Path(testing_dir, "tests", "unit", "schema", "testdata"))
+
+    def test_samconfig_validates_against_schema(self):
+        self.assertIsNotNone(self.schema, "Schema was not set")
+
+        passing_tests_dir = Path(self.testdata_dir, "passing_tests")
+        tests_not_passing = []
+
+        # Read in and assert all files in passing_tests pass
+        for config_file_path in passing_tests_dir.iterdir():
+            config_file = FILE_MANAGER_MAPPER[config_file_path.suffix].read(Path(str(passing_tests_dir), config_file_path))
+            self.assertNotEqual(config_file, {}, f"Config file {config_file_path} should be read correctly")
+
+            try:
+                jsonschema.validate(config_file, self.schema)
+            except jsonschema.ValidationError as e:
+                tests_not_passing.append(f"<{config_file_path}: {e.message}>")
```

Comment:
Should we stop the test immediately instead of appending to an error list (failing fast)?

Reply:
Yup we definitely could. I implemented it this way so we could see if multiple tests were failing for the same or different reasons, but keeping things quicker may be ideal instead 👍